package/MDAnalysis/coordinates/DCD.py

```diff
@@ -462,20 +462,9 @@ def _dcd_header(self):
             'charmm', 'first', 'with_unitcell']
         return dict(zip(desc, struct.unpack("LLiiiiidiPPiiii", self._dcd_C_str)))
 
-    def __iter__(self):
-        # Reset the trajectory file, read from the start
-        # usage is "from ts in dcd:" where dcd does not have indexes
+    def _reopen(self):
         self._reset_dcd_read()
 
-        def iterDCD():
-            for i in xrange(0, self.numframes, self.skip):  # FIXME: skip is not working!!!
```

Comment:
Looking at the DCDReader code, we just set `self.skip = 1`. 
I think it really ought be called "step" but in the underlying dcd.c code, it's called `skip`. In principle the DCD C code should allow frame skipping at the C level with `skip_dcdstep()` but I just found the comment *"XXX this needs to be changed"*....
```C
    // Figure out how many steps to skip
    numskip = skip - (dcd->setsread % skip) - 1;
    /* XXX this needs to be changed */
    rc = skip_dcdstep(dcd->fd, dcd->natoms, dcd->nfixed, dcd->charmm, numskip);
```
It might be related to having the possibility of fixed atoms in a DCD or it might simply not work as expected and that's why we seem to have effectively disabled skipping at the C level. I'm sorry, I don't remember.

It seems to work, though, because:
```python
import MDAnalysis; from MDAnalysis.tests.datafiles import PSF, DCD
u = MDAnalysis.Universe(PSF,DCD)
u.trajectory.skip = 10
frames = [ts.frame for ts in u.trajectory]
print(frames)
```
gives
```
[10, 20, 30, 40, 50, 60, 70, 80, 90]
```
as expected,